### PR TITLE
Initial form components documentation

### PR DESCRIPTION
**Pull request by jpzwarte**

### CODE REVIEW

File: website/src/categories/components/form/code.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+eleventyNavigation:
+  parent: Form
+  key: FormCode
+---
+
+The `<sl-form>` component fulfills three functions:
+1. It provides a default layout for form fields.
+2. It manages the way required/optional fields should be marked.
+2. It provides a way to validate all form fields at once.
+3. It allow you to query the state of the form.
```
Comment (Diaan) [inline website/src/categories/components/form/code.md:16]:
```suggestion
3. It allows you to query the state of the form.
```

File: website/src/categories/components/form/code.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+## Layout
+
+By default, the form component has a vertical flexbox layout. This means that form fields will stack on top of each other. If you want to change this, you can use a grid layout instead.
+
+```css
+/* Define a two column grid layout. */
```
Comment (Diaan) [inline website/src/categories/components/form/code.md:23]: the colour of the comment text is pretty hard to read on the grey background

File: website/src/categories/components/form/code.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    // Submit the form
+  }
+});
+```
+
+## State
+
+You can query the state of the entire form by simply getting the `valid` property on the `<sl-form>`. This will return `false` if any form field within the form is not valid.
+
+{% include "../component-table.njk" %}
```
Comment (Diaan) [inline website/src/categories/components/form/code.md:54]: for some reason the "Attribute" column is empty in the properties table.

File: website/src/categories/components/form-field/usage.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  <sl-form-field hint="Please enter a descriptive name for the course." label="Course name">
+    <sl-text-field name="courseName"></sl-text-field>
+  </sl-form-field>
+  ```
+
+</div>
+
+</section>
+
+If a field is part of a larger form, you should wrap it in a form field component. This will ensure that the field is properly labeled and that the hint is announced to screen readers.
```
Comment (Diaan) [inline website/src/categories/components/form-field/usage.md:31]: maybe use the `When to use` and `When not to use` structure here as well?

Comment (Diaan) [reply website/src/categories/components/form-field/usage.md:31]: Also, link to the form component documentation page?

File: website/src/categories/components/form-field/usage.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+A form field should always have a label. That label can be hidden from view, but it should always be present in the DOM. The label is used to associate the form field with the form control. The label is also used to announce the form field to screen readers.
+
+The hint is optional. It is used to provide additional information about the form field. The hint is announced to screen readers after the label.
+
+Both label and hint are accessible as properties, but you can also use their respective slots if you need more control over the content.
+
+## Validation messages
+
+The form field component listens for `sl-update-validity` events from the form control. When the form control is invalid, the form field will display the `validationMessage`. When the form control is valid, the form field will hide the error message.
```
Comment (Diaan) [inline website/src/categories/components/form-field/usage.md:54]: i would move this part to the "code" page, the "usage" is usually geared more towards designers.

File: website/src/categories/components/form-field/usage.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+The API tried to closely follow the native `ValidityState` API. The `validationMessage` property is a string that contains the error message. The `validity` property is an object that contains the following properties.
+
+You can set a custom validity by calling the `setCustomValidity(message)` method on the form control. This will trigger the `sl-update-validity` event on the form control.
+
+The `reportValidity()` method can be called on the form field to trigger validation on the form control. This method returns a boolean indicating whether the form control is valid or not. Normally you would not call this method directly, but instead call it on the parent `<sl-form>` component.
+
+## Integration
+
+Besides delegating the `reportValidity()` calls from the parent `<sl-form>` to the form control, the form field also registers itself with the parent `<sl-form>`. This allows the `<sl-form>` to keep track of all the form fields and their validity. This done by dispatching the `sl-form-field` event for which the `<sl-form>` listens.
```
Comment (Diaan) [inline website/src/categories/components/form-field/usage.md:64]: same; move to code tab

File: website/src/categories/components/form/usage.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+The form component should be used whenever you have a form that needs to be filled out by a user. The form component is a container for form fields.
+
+## When not to use
+
+Do not use the form component if you only have one form field. Usually this indicates a specific usage of a form control, and should be used on its own.
+
+## Required or optional
+
+The form itself manages a `mark` state. This determines how required or optional labels should be shown. If the required fields outnumber the optional fields, only the optional fields will be marked. If it is the other way around, then only the required fields will be marked.
```
Comment (Diaan) [inline website/src/categories/components/form/usage.md:71]: maybe this is also more suitable for the "code" page?

File: website/src/categories/components/form/accessibility.md [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+title: Form accessibility
+tags: accessibility
+eleventyNavigation:
+  parent: Form
+  key: FormAccessibility
+---
+
+<section>
+
+Lorem
```
Comment (anna-lach) [inline website/src/categories/components/form/accessibility.md:11]: I think we need some information here instead of `Lorem` if we want to show it live on the website.

File: website/src/categories/components/form/usage.md (hunk 7 above)
Comment (jpzwarte) [reply website/src/categories/components/form/usage.md:71]: I think this section is more about the behavior (usage) than actual code. So I'd rather keep it here.

**Comment by DanielleRameau** [comment]
Screenshot 2024-02-14 at 10 17 30
@jpzwarte  Some changes for the code block:
background: #000D33 (primary.900)
token.comment: #E0EAFF (accent.100)
token.selector and token.property:
 #5985FF (accent.200)
token.punctuation: white

File: website/src/categories/components/form/code.md [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+title: Form code
+tags: code
+APIdescription: {
+  sl-form: "Form component is meant as a wrapper around form fields."
+}
+eleventyNavigation:
+  parent: Form
+  key: FormCode
+---
+
```
Comment (anna-lach) [inline website/src/categories/components/form/code.md:11]: Can you please add `installation` section like in the switch PR #953

File: website/src/ts/scripts/tabs.ts [hunk 10]
```diff
@@ -328,7 +328,7 @@ function selectTab(tab: Element): void {
     observer.observe(tabsContainer as Element);
   }
 
-  const tabSections = tabContent?.querySelectorAll('section[id]');
+  const tabSections = tabContent?.querySelectorAll('h2[id]');
```
Comment (anna-lach) [inline website/src/ts/scripts/tabs.ts:331]: Can you please revert this change and leave it with sections for now?

File: website/src/categories/components/form/code.md (hunk 3 above)
Comment (jpzwarte) [reply website/src/categories/components/form/code.md:107]: That's because none of these are available as attributes :)

**Review by anna-lach — APPROVED** [review]
